src/model/guild/member.rs

```diff
@@ -325,6 +325,37 @@ impl Member {
     /// [Kick Members]: ../permissions/struct.Permissions.html#associatedconstant.KICK_MEMBERS
     #[cfg(feature = "http")]
     pub fn kick(&self, cache_http: impl CacheHttp) -> Result<()> {
+        self.kick_with_reason(cache_http, None)
+    }
+
+    /// Kicks the member from the guild, with a reason.
+    ///
+    /// **Note**: Requires the [Kick Members] permission.
+    ///
+    /// # Examples
+    ///
+    /// Kicks a member from it's guild, with an optional reason:
+    ///
+    /// ```rust,ignore
+    /// match member.kick(&ctx.http, Some("Reasons")) {
+    ///     Ok(()) => println!("Successfully kicked member"),
+    ///     Err(Error::Model(ModelError::GuildNotFound)) => {
+    ///         println!("Couldn't determine guild of member");
+    ///     },
+    ///     Err(Error::Model(ModelError::InvalidPermissions(missing_perms))) => {
+    ///         println!("Didn't have permissions; missing: {:?}", missing_perms);
+    ///     },
+    ///     _ => {},
+    /// }
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Same as [`kick`]
+    ///
+    /// [`kick`]: #method.kick
+    #[cfg(feature = "http")]
+    pub fn kick_with_reason(&self, cache_http: impl CacheHttp, reason: Option<&str>) -> Result<()> {
```

Comment:
I don't know if it's just me but it feels weird to have a method named kicked_with_reason but making the reason optional. It kind of seems to counteract what the method name is wanting.

Reply:
Did this so Args.remains() could be used directly as a reason

Reply:
I agree with Novus. If the method tells its purpose is the ability to kick someone from a guild with a reason as an explanation why, it should be a given that it will always expect a reason from the user. It should never be `None`.

This should apply to the other `kick`s you have changed. I would like to relieve the library from breaking the interface too much for something as simple as kicking with a reason.